crates/ra_assists/src/handlers/fill_match_arms.rs

```diff
@@ -104,6 +126,22 @@ fn resolve_enum_def(sema: &Semantics<RootDatabase>, expr: &ast::Expr) -> Option<
     })
 }
 
+fn resolve_tuple_of_enum_def(
+    sema: &Semantics<RootDatabase>,
+    expr: &ast::Expr,
+) -> Option<Vec<hir::Enum>> {
+    Some(
+        sema.type_of_expr(&expr)?
+            .tuple_fields(sema.db)
+            .iter()
+            .map(|ty| match ty.as_adt() {
+                Some(Adt::Enum(e)) => e,
+                _ => panic!("handle the case of tuple containing non-enum"),
```

Comment:
I don't think panicking during assist construction just because we did not implement something is a good idea: first of all, it pollutes the logs with big traces that do not say anything meaningful besides that the developers did not cover all cases (and this might be frequent, since this construction also happens on hover). 
Also, with this panic, calling the `Quick fix` shortcut will show endless progress bar in VS Code, which is a bit distracting :)

So I propose to simply match the only case here and maybe leave a comment about the rest of the cases.

Reply:
Oops, I mean to go back and fix that. Fixed.

Reply:
Note that you can use `todo!()` or a `//TODO` comment for such cases: CI would prevent mering code with todos into master.